src/components/Tabs.tsx

```diff
@@ -0,0 +1,47 @@
+import classNames from 'classnames';
+import { Link, useParams } from 'react-router-dom';
+import { Tab } from '../types/Tab';
+
+type Props = {
+  tabs: Tab[],
+};
+
+export const Tabs: React.FC<Props> = ({ tabs }) => {
+  const { tabId } = useParams();
+  const selectedTab = tabs.find(tab => tab.id === tabId) || null;
+
+  return (
+    <>
+      {tabs.some(tab => tab.id === tabId) || tabId === undefined
+        ? (
+          <>
+            <h1 className="title">Tabs page</h1>
+            <div className="tabs is-boxed">
+              <ul>
+                {tabs.map(tab => (
+                  <li
+                    key={tab.id}
+                    data-cy="Tab"
+                    className={classNames({
+                      'is-active': tab.id === tabId,
+                    })}
+                  >
+                    <Link to={`/tabs/${tab.id}`}>
+                      {tab.title}
+                    </Link>
+                  </li>
+                ))}
+              </ul>
+            </div>
+            <div className="block" data-cy="TabContent">
+              {selectedTab
+                ? selectedTab.content
+                : 'Please select a tab'}
+            </div>
+          </>
+        ) : (
+          <h1 className="title">Not Found</h1>
+        )}
```

Comment:
I would make this check while building route tree
(please check my tips about App component)